Source/Session.swift

```diff
@@ -72,9 +126,41 @@ open class Session {
+    ///
+    /// - Parameters:
+    ///   - configuration:            `URLSessionConfiguration` to be used to create the underlying `URLSession`. Changes
+    ///                               to this value after being passed to this initializer will have no effect.
+    ///                               `URLSessionConfiguration.af.default` by default.
+    ///   - delegate:                 `SessionDelegate` that handles `session`'s delegate callbacks as well as `Request`
+    ///                               interaction. `SessionDelegate()` by default.
+    ///   - rootQueue:                Root `DispatchQueue` for all internal callbacks and state updats. **MUST** be a
+    ///                               serial queue. `DispatchQueue(label: "org.alamofire.session.rootQueue")` by default.
+    ///   - startRequestsImmediately: Determines whether this instance will automatically start all `Request`s. `true`
+    ///                               by default. If set to `false`, all `Request`s created must have `.resume()` called.
+    ///                               on them for them to start.
+    ///   - requestQueue:             `DispatchQueue` on which to perform `URLRequest` creation. By default this queue
+    ///                               will use the `rootQueue` as its `target`. A separate queue can be used if it's
+    ///                               determined request creation is a bottleneck, but that should only be done after
+    ///                               careful testing and profiling. `nil` by default.
+    ///   - serializationQueue:       `DispatchQueue` on which to perform all response serialization. By default this
+    ///                               queue will use the `rootQueue` as its `target`. A separate queue can be used if
+    ///                               it's determined response serialization is a bottleneck, but that should only be
+    ///                               done after careful testing and profiling. `nil` by default.
+    ///   - interceptor:              `RequestInterceptor` to be used for all `Request`s created by this instance. `nil`
+    ///                               by default.
+    ///   - serverTrustManager:       `ServerTrustManager` to be used for all trust evaluations by this instance. `nil`
+    ///                               by default.
+    ///   - redirectHandler:          `RedirectHandler` to be used by all `Request`s created by this instance. `nil` by
+    ///                               default.
+    ///   - cachedResponseHandler:    `CachedResponseHandler` to be used by all `Request`s created by this instance.
+    ///                               `nil` by default.
+    ///   - eventMonitors:            Additional `EventMonitor`s used by the instance. Alamofire always adds a
+    ///                               `AlamofireNotifications` `EventMonitor` to the array passed here. `[]` by default.
```

Comment:
This makes me wonder if `AlamofireNotifications` should be an internal `EventMonitor` and the fact that we're doing this is just an internal detail that shouldn't be noted in the docstrings. I'm also not sure that the `CompositeEventHandler` looks for duplicates. For example, if the `AlamofireNotifications` is `public`, is it intended to be passed in? Would we then have two of them running? Thoughts?

Reply:
My original thought here is that users may want to be able to create `Session`s without `AlamofireNotifications`, but I never got around to working out how to expose such functionality. If we don't want that capability, we could just make `AlamofireNotifications` an internal implementation detail.

Reply:
Gotcha. I don't really see that being an issue myself and lean towards making them internal altogether. If we end up with requests to be able to disable that functionality, then we could always add that option in the future. Maybe follow that change up in a separate PR?